internal/output/markdown.go

```diff
@@ -121,15 +121,15 @@ type MarkdownCtx struct {
 	MarkdownOptions              MarkdownOptions
 }
 
-func ToMarkdown(out Root, opts Options, markdownOpts MarkdownOptions) ([]byte, error) {
+func ToMarkdown(out Root, opts Options, markdownOpts MarkdownOptions) ([]byte, error, int, int) {
```

Comment:
convention is to always put the error as the last return arg, let's change it to that.

I'm also not a massive fan of having a ton of these return args. Can we change it so we return a struct:

```
type MarkdownOutput struct {
     Msg []byte
     RuneLen  int
     OriginalMsgSize int
}
```

This way it's a bit more readable from a consumer stand point and we don't have to have a ton of return args.